From 0bd845490befbe367ae1eb051ebbe3f844ebf472 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 7 Mar 2025 17:45:46 +0100
Subject: [PATCH 1/2] deps: Build GLEW with EGL support

EGL support is necessary for offscreen device access, and
eventually Wayland support.
---
 deps/GLEW/glew/CMakeLists.txt | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/deps/GLEW/glew/CMakeLists.txt b/deps/GLEW/glew/CMakeLists.txt
index 513ee4ebb512..4aa10c918812 100644
--- a/deps/GLEW/glew/CMakeLists.txt
+++ b/deps/GLEW/glew/CMakeLists.txt
@@ -3,12 +3,9 @@ project(GLEW)
 
 find_package(OpenGL REQUIRED)
 
-# we do not support wayland for now
-if(NOT CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    if(OpenGL_EGL_FOUND)
-        message(STATUS "building GLEW for EGL (hope that wxWidgets agrees, otherwise you won't have any output!)")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DGLEW_EGL")
-    endif()
+if(OpenGL_EGL_FOUND)
+    message(STATUS "building GLEW for EGL (hope that wxWidgets agrees, otherwise you won't have any output!)")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DGLEW_EGL")
 endif()
 
 add_library(GLEW src/glew.c)
-- 
2.50.0


From 17d096e755f38cb73d2f8c40a8cc693bac386225 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 15 Aug 2025 18:04:23 +0200
Subject: [PATCH 2/2] main: Remove OSMesa dependency

The "offscreen" rendering code was supposed to use OSMesa's offscreen
buffers so as to avoid an X11 dependency, but the CLI::run() function
called XInitThreads() every time, so there's very little chance that
this ever worked.

Remove the dependency so that we don't depend on functionality that's
been removed from newer mesa:
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/33836

Closes: https://github.com/bambulab/BambuStudio/issues/7017
Closes: https://github.com/bambulab/BambuStudio/issues/6951
---
 Containerfile             | 1 -
 Dockerfile                | 1 -
 linux.d/debian            | 1 -
 linux.d/fedora            | 1 -
 src/BambuStudio.cpp       | 8 --------
 src/slic3r/CMakeLists.txt | 2 +-
 6 files changed, 1 insertion(+), 13 deletions(-)

diff --git a/Containerfile b/Containerfile
index a44215537e6f..1726adc93895 100644
--- a/Containerfile
+++ b/Containerfile
@@ -44,7 +44,6 @@ RUN apt-get update && apt-get install  -y \
     libgtk-3-dev libwebkit2gtk-4.0-dev \
     libsoup2.4-dev \
     libgstreamer1.0-dev libgstreamer-plugins-good1.0-dev libgstreamer-plugins-base1.0-dev libgstreamerd-3-dev \
-    libosmesa6-dev \
     libssl3 libssl-dev libcurl4-openssl-dev libsecret-1-dev \
     libudev-dev \
     curl \
diff --git a/Dockerfile b/Dockerfile
index 878d213f2547..58df76ba7a2e 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -32,7 +32,6 @@ RUN apt-get update && apt-get install  -y \
     libgstreamer-plugins-good1.0-dev \
     libgtk-3-dev \
     libgtk-3-dev \
-    libosmesa6-dev \
     libsecret-1-dev \
     libsoup2.4-dev \
     libssl3 \
diff --git a/linux.d/debian b/linux.d/debian
index c076b25e23d3..44aad6269ed4 100644
--- a/linux.d/debian
+++ b/linux.d/debian
@@ -12,7 +12,6 @@ REQUIRED_DEV_PACKAGES=(
     wget
     libgstreamerd-3-dev
     libsecret-1-dev
-    libosmesa6-dev
     libssl-dev
     eglexternalplatform-dev
     libcurl4-openssl-dev
diff --git a/linux.d/fedora b/linux.d/fedora
index cae064e15e1c..3004b832593a 100644
--- a/linux.d/fedora
+++ b/linux.d/fedora
@@ -23,7 +23,6 @@ REQUIRED_DEV_PACKAGES=(
     libtool
     m4
     mesa-libGLU-devel
-    mesa-libOSMesa-devel
     mesa-libGL-devel
     ninja-build
     openssl-devel
diff --git a/src/BambuStudio.cpp b/src/BambuStudio.cpp
index 79cd8b958274..4f08a0f3270d 100644
--- a/src/BambuStudio.cpp
+++ b/src/BambuStudio.cpp
@@ -5510,10 +5510,6 @@ int CLI::run(int argc, char **argv)
             glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_COMPAT_PROFILE);
 #endif
 
-#ifdef __linux__
-            glfwWindowHint(GLFW_CONTEXT_CREATION_API, GLFW_OSMESA_CONTEXT_API);
-#endif
-
             GLFWwindow* window = glfwCreateWindow(640, 480, "base_window", NULL, NULL);
             if (window == NULL)
             {
@@ -6666,10 +6662,6 @@ int CLI::run(int argc, char **argv)
                 glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_COMPAT_PROFILE);
 #endif
 
-#ifdef __linux__
-                glfwWindowHint(GLFW_CONTEXT_CREATION_API, GLFW_OSMESA_CONTEXT_API);
-#endif
-
                 GLFWwindow* window = glfwCreateWindow(640, 480, "base_window", NULL, NULL);
                 if (window == NULL)
                 {
diff --git a/src/slic3r/CMakeLists.txt b/src/slic3r/CMakeLists.txt
index d21209d28a9d..2f0b95badd92 100644
--- a/src/slic3r/CMakeLists.txt
+++ b/src/slic3r/CMakeLists.txt
@@ -660,9 +660,9 @@ elseif (CMAKE_SYSTEM_NAME STREQUAL "Linux")
     FIND_LIBRARY(WAYLAND_EGL_LIBRARIES    NAMES wayland-egl)
     FIND_LIBRARY(WAYLAND_CLIENT_LIBRARIES NAMES wayland-client)
     find_package(CURL REQUIRED)
-    target_link_libraries(libslic3r_gui ${DBUS_LIBRARIES} OSMesa)
     target_link_libraries(libslic3r_gui
         OpenGL::EGL
+        ${DBUS_LIBRARIES}
         ${WAYLAND_SERVER_LIBRARIES}
         ${WAYLAND_EGL_LIBRARIES}
         ${WAYLAND_CLIENT_LIBRARIES}
-- 
2.50.0

